packages/amplify_auth_cognito/lib/src/CognitoSignUp/cognito_user_attributes.dart

```diff
@@ -1,24 +1,185 @@
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:flutter/foundation.dart';
+
 /// Standard user attributes available for configuring via `Amplify.Auth.signUp`,
 /// or updating via `Amplify.Auth.updateUserAttribute` and
 /// `Amplify.Auth.updateUserAttributes`.
 ///
 /// More information about these attributes can be found
 /// [here](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html).
-abstract class CognitoUserAttributes {
-  static const address = 'address';
-  static const email = 'email';
-  static const familyName = 'family_name';
-  static const gender = 'gender';
-  static const givenName = 'given_name';
-  static const locale = 'locale';
-  static const middleName = 'middle_name';
-  static const name = 'name';
-  static const nickname = 'nickname';
-  static const phoneNumber = 'phone_number';
-  static const picture = 'picture';
-  static const preferredUsername = 'preferred_username';
-  static const profile = 'profile';
-  static const updatedAt = 'updated_at';
-  static const website = 'website';
-  static const zoneinfo = 'zoneinfo';
+@immutable
+class CognitoUserAttributes extends UserAttributeKey {
+  const CognitoUserAttributes._(this.key, {this.readOnly = false});
+
+  /// Prefix for custom Cognito attributes.
+  static const _customPrefix = 'custom:';
+
+  /// The JSON key for this attribute.
+  @override
+  final String key;
+
+  /// Whether this attribute can only be read from Cognito.
+  final bool readOnly;
+
+  /// Creates a custom Cognito attribute.
+  const CognitoUserAttributes.custom(String key)
+      : key = '$_customPrefix$key',
+        readOnly = false;
+
+  /// Parses the given Cognito attribute key.
+  factory CognitoUserAttributes.parse(String key) {
+    key = key.toLowerCase();
+    return values.firstWhere(
+      (attr) => attr.key == key,
+      orElse: () {
+        if (!key.startsWith(_customPrefix)) {
```

Comment:
This is throwing when `fetchUserAttributes()` is called because there is an attribute returned with a key of `sub`.

Reply:
Good catch @Jordan-Nelson